lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -76,6 +77,16 @@ static FIRRTLType getFIRRTLType(Type type) {
         unsigned width = indexType.kInternalStorageBitWidth;
         return UIntType::get(context, width);
       })
+      .Case<TupleType>([&](TupleType tupleType) -> FIRRTLType {
+        using BundleElement = BundleType::BundleElement;
+        llvm::SmallVector<BundleElement, 2> elements;
+        unsigned fieldCnt = 0;
+        for (auto elementType : tupleType.getTypes())
+          elements.emplace_back(BundleElement(
+              StringAttr::get(context, llvm::formatv("field{0}", fieldCnt++)),
```

Comment:
nit: we generally prefer pre-increment according to LLVM style guidelines.

Reply:
If you are fine with the fields being index from 1 to N instead of 0 to N - 1 I can change that. 

IIRC, post-increment is mainly problematic for iterators that might keep state.

Reply:
Changing to enumerate, thus this is resolved.